### ISSUE TITLE
[Issue]: Upscale model suddenly stops working

### ISSUE THREAD

**Issue by Woisek** [vladmandic/automatic#391]
### Issue Description

Idk if this is automatic related, but I have to start somewhere:
Some time ago, I downloaded some Swin2SR upscalers into my SwinIR folder. 

Swin2SR_ClassicalSR_X2_64.pth
Swin2SR_ClassicalSR_X4_64.pth
Swin2SR_Lightweight_X2_64.pth
Swin2SR_RealworldSR_X4_64_BSRGAN_PSNR.v2.pth

Today, I did a bunch of wallpaper upscales useing the Swin2SR_ClassicalSR_X2_64.pth and Swin2SR_ClassicalSR_X4_64.pth, working with great results.
After a TV break of maybe 1½ hours, I wanted to continou, but suddenly, I get an error:

100%|██████████████████████████████████████████████████████████████████████████████████| 40/40 [00:05<00:00,  6.67it/s]
[Tiled VAE]: the input size is tiny and unnecessary to tile.
gradio call: KeyError
┌───────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────┐
│ I:\Super SD 2.0\automatic\modules\call_queue.py:58 in f                                                             │
│                                                                                                                     │
│    57 │   │   │   │   pr.enable()                                                                                   │
│ >  58 │   │   │   res = list(func(*args, **kwargs))                                                                 │
│    59 │   │   │   if shared.cmd_opts.profile:                                                                       │
│                                                                                                                     │
│ I:\Super SD 2.0\automatic\modules\call_queue.py:38 in f                                                             │
│                                                                                                                     │
│    37 │   │   │   try:                                                                                              │
│ >  38 │   │   │   │   res = func(*args, **kwargs)                                                                   │
│    39 │   │   │   finally:                                                                                          │
│                                                                                                                     │
│                                               ... 7 frames hidden ...                                               │
│                                                                                                                     │
│ I:\Super SD 2.0\automatic\extensions-builtin\SwinIR\scripts\swinir_model.py:41 in do_upscale                        │
│                                                                                                                     │
│    40 │   def do_upscale(self, img, model_file):                                                                    │
│ >  41 │   │   model = self.load_model(model_file)                                                                   │
│    42 │   │   if model is None:                                                                                     │
│                                                                                                                     │
│ I:\Super SD 2.0\automatic\extensions-builtin\SwinIR\scripts\swinir_model.py:93 in load_model                        │
│                                                                                                                     │
│    92 │   │   if params is not None:                                                                                │
│ >  93 │   │   │   model.load_state_dict(pretrained_model[params], strict=True)                                      │
│    94 │   │   else:                                                                                                 │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
KeyError: 'params_ema'

And I get this for all models except the last one, Swin2SR_RealworldSR_X4_64_BSRGAN_PSNR.v2.pth. This model still works. I'm confused, because I don't unterstand how this could be, that something that works an hour before, suddenly don't anymore, without any updates to WebUI applied or such.

Any suggestions? 😌

### Version Platform Description

Win10, Firefox 106
22:25:00-319643 INFO     Version: 4cb46d54 Sat Apr 22 13:33:11 2023 -0400

**Comment by vladmandic** [comment]
this line `gradio call: KeyError` points me towards #224 - if that is true, lets close this.

**Comment by Woisek** [comment]
Well, yes, I experience also double entries of the upscaler, but in the meantime I have restarted (= closed the cmd.exe and started new) the WebUI so many times now and this issue still exists., Total closing can't be the solution, because .. it don't work (at least here it doesn't). 😔

**Comment by vladmandic** [comment]
yeah, took a closer look, this is not about restart server from ui.
torch loaded a model, but that model did not have expected property. which makes no sense, but there is no info logged about it, so not much i can do. i've added model load info so we can see what is actually getting loaded. update and post traceback when it happens again.

**Comment by Woisek** [comment]
OK, will watch it.

**Comment by Woisek** [comment]
Loading weights: I:\Super SD 2.0\automatic\models\SwinIR\Swin2SR_ClassicalSR_X2_64.pth ------------ 0.0/68.8 MB -:--:--
gradio call: KeyError
┌───────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────┐
│ I:\Super SD 2.0\automatic\modules\call_queue.py:58 in f                                                             │
│                                                                                                                     │
│    57 │   │   │   │   pr.enable()                                                                                   │
│ >  58 │   │   │   res = list(func(*args, **kwargs))                                                                 │
│    59 │   │   │   if shared.cmd_opts.profile:                                                                       │
│                                                                                                                     │
│ I:\Super SD 2.0\automatic\modules\call_queue.py:38 in f                                                             │
│                                                                                                                     │
│    37 │   │   │   try:                                                                                              │
│ >  38 │   │   │   │   res = func(*args, **kwargs)                                                                   │
│    39 │   │   │   finally:                                                                                          │
│                                                                                                                     │
│                                               ... 7 frames hidden ...                                               │
│                                                                                                                     │
│ I:\Super SD 2.0\automatic\extensions-builtin\SwinIR\scripts\swinir_model.py:40 in do_upscale                        │
│                                                                                                                     │
│    39 │   def do_upscale(self, img, model_file):                                                                    │
│ >  40 │   │   model = self.load_model(model_file)                                                                   │
│    41 │   │   if model is None:                                                                                     │
│                                                                                                                     │
│ I:\Super SD 2.0\automatic\extensions-builtin\SwinIR\scripts\swinir_model.py:93 in load_model                        │
│                                                                                                                     │
│    92 │   │   if params is not None:                                                                                │
│ >  93 │   │   │   model.load_state_dict(pretrained_model[params], strict=True)                                      │
│    94 │   │   else:                                                                                                 │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
KeyError: 'params_ema'

😔

**Comment by vladmandic** [comment]
i'm not familiar with `Swin2SR_ClassicalSR_X2_64`, but i'll test it. where did you download it from?
how about standard `SwinIR_4x`?

**Comment by Woisek** [comment]
It's from the official? upscaler wiki -> https://upscale.wiki/wiki/Model_Database

>how about standard SwinIR_4x?

In therms of functionality: This one works. It comes with the repo, I think. All the upsacler I have are working, just those three do not. But they have worked before, so I can rule out defect or destroyded files.

**Comment by vladmandic** [comment]
i understand, but now i finally have something to work with. although it may be few days until i get to it.

**Comment by vladmandic** [comment]
@Woisek i don't see `Swin2SR_ClassicalSR_X2_64` in the wiki model database list - can you post actual link to the model?

**Comment by Woisek** [comment]
Sorry, I go it from elsewhere - https://github.com/mv-lab/swin2sr
Here is something on Huggingface -> https://huggingface.co/spaces/jjourney1125/swin2sr
Hope that helps.

**Comment by Woisek** [comment]
Sorry, I downloaded those from GitHub -> https://github.com/mv-lab/swin2sr/releases

**Comment by Woisek** [comment]
Hmm ... so how come that 1 of them works, while the others are not? And why did they work before?
@ contribute: Do you mean, I should alter the filename and test it?

**Comment by vladmandic** [comment]
re: how come that 1 of them works
why not? maybe they do, maybe they don't. it all depends how close they are in their architecture to original

re: contribute
no, its not about altering filename and testing - if model doesn't work, but its origin is swinir original model, then maybe it can be made to work by extending the code i've posted above to account for that model architecture (sample sizes, etc.).
